tests/ordering/tests.py

```diff
@@ -484,3 +486,16 @@ def test_order_by_ptr_field_with_default_ordering_by_expression(self):
         ca4 = ChildArticle.objects.create(headline='h1', pub_date=datetime(2005, 7, 28))
         articles = ChildArticle.objects.order_by('article_ptr')
         self.assertSequenceEqual(articles, [ca4, ca2, ca1, ca3])
+
+    def test_group_by_does_not_contains_meta_ordering(self):
+        Article.objects.filter(headline='Article 1').update(author=self.author_1)
+        Article.objects.filter(headline='Article 2').update(author=self.author_1)
+        Article.objects.filter(headline='Article 3').update(author=self.author_2)
+        Article.objects.filter(headline='Article 4').update(author=self.author_2)
+
+        articles = (OrderedByPK.objects
+                               .values('author')
+                               .annotate(count=Count('author')))
+
+        self.assertEquals(len(articles), articles.count())
+        self.assertEquals(articles.count(), 2)
```

Comment:
You can reuse `Article`, e.g.
```suggestion
        Article.objects.filter(headline='Article 1').update(author=self.author_1)
        Article.objects.filter(headline='Article 2').update(author=self.author_1)
        Article.objects.filter(headline='Article 3').update(author=self.author_1)
        Article.objects.filter(headline='Article 4').update(author=self.author_2)
        articles = Article.objects.values('author').annotate(count=Count('author'))
        self.assertCountEqual(articles, [
            {'author': self.author_1.pk, 'count': 3},
            {'author': self.author_2.pk, 'count': 1},
        ])
```